src/lib.rs

```diff
@@ -130,24 +135,49 @@ pub extern "C" fn rustls_client_session_new(
             return rustls_result::ERROR;
         }
     };
-    let client = ClientSession::new(&config, name_ref);
+    let session = ClientSession::new(&config, name_ref);
 
-    // We've succeeded. Put the client on the heap, and transfer ownership
-    // to the caller. After this point, we must return CRUSTLS_OK so the
-    // caller knows it is responsible for this memory.
-    let b = Box::new(client);
+    // We've succeeded. Put a rustls_client_session on the heap, and transfer
+    // ownership to the caller. After this point, we must return CRUSTLS_OK so
+    // the caller knows it is responsible for this memory.
+    let b = Box::new(rustls_client_session {
+        session,
+        error: "".to_string(),
+    });
     unsafe {
         *session_out = Box::into_raw(b) as *mut _;
     }
 
     return rustls_result::OK;
 }
 
+const NULL_SESSION: &str = "session was NULL";
+
+/// After a rustls_client_session method returns an error, you may call
+/// this method to get a pointer to a buffer containing a detailed error
+/// message. The contents of the error buffer will by len bytes long,
+/// UTF-8 encoded, and not NUL-terminated.
```

Comment:
This should clearly indicate that the pointer is invalid after you call any other method with the `rustls_client_session`.

Reply:
Also, what are the thread safety implications here? Was it already the case that a `ClientSession` can't be shared between threads?